File: sqlite3/ocaml/read/sqlite3_read.ml
(* ------------------------------------------------------------------- *)
(*
	sqlite3_update.ml

					Dec/24/2010

*)
(* ------------------------------------------------------------------- *)
open Printf
open Sqlite3

let file_in = Sys.argv.(1)
let _ =
	print_endline "*** 開始 ***";
	printf "%s\n" file_in;
	print_endline "*** pppp ***";

	let db = db_open file_in in
	print_endline "*** qqqq ***";
	ignore(exec_not_null db 
		(fun row headers ->
			Array.iter(fun el -> Printf.printf "%s\t" el) row;
			Printf.printf "\n";
		) "SELECT id,name,population,date_mod FROM cities"); 
	print_endline "*** 終了 ***";
	db_close db;

(* ------------------------------------------------------------------- *)

File: sqlite3/go/read/sqlite3_read.go
// ----------------------------------------------------------------
//
//	sqlite3_read.go
//
//					May/20/2018
//
// ----------------------------------------------------------------
package main

import (
	"fmt"
	"database/sql"
	_ "github.com/mattn/go-sqlite3"
)

// ----------------------------------------------------------------
func main() {
	fmt.Println ("*** 開始 ***")

	dict_aa := make (map[string](map[string]interface{}))

	db_file := "/var/tmp/sqlite3/cities.db"
	db, err := sql.Open("sqlite3", db_file)
	if err != nil {
		fmt.Println ("*** error ***")
		fmt.Println(err)
		return
		}

	sql_str := "select id,name,population,date_mod from cities"

	rows, err := db.Query (sql_str)
	if err != nil {
		fmt.Println(err)
	}
	defer rows.Close ()

	for rows.Next() {
	var id string
	var name string 
	var population int
	var date_mod string 
	if err := rows.Scan(&id,&name,&population,&date_mod); err != nil {
		fmt.Println(err)
		}

	dict_aa[id] = unit_gen_proc (name,population,date_mod)
	}

	dict_display_proc (dict_aa)

	fmt.Println ("*** 終了 ***")
}

// ----------------------------------------------------------------
